### PR TITLE
[PWGCF] Updated CorrSparse with added histograms and a changed dPhi definition

### PR DESCRIPTION
**Pull request by Thorkj**

### CODE REVIEW

**Review by victor-gonzalez — APPROVED** [review]
Please, for the next iteration consider the `const-ref-in-process`, the `const-ref-in-for-loop`, and the using `PI` constant and their multiples/fractions from `o2::constants::math` `lint` suggestions

**Comment by Thorkj** [comment]
I have troubleshooted the lint errors and can't find the solution, it seems to be the same setup as the previous accepted file. For some reason it causes issues now. Please let me know if you have any suggestions.

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 1]
```diff
@@ -33,12 +37,8 @@ struct CorrSparse {
   Configurable<float> cfgPtCutMin = {"minpt", 0.2, "Minimum accepted track pT. Default 0.2 GeV"};
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:37]: For this one, for example, you have to change it to
```
Configurable<float> cfgPtCutMin = {"cfgPtCutMin", 0.2, "Minimum accepted track pT. Default 0.2 GeV"};
```
to get rid of the error in that line
Similar for the similar errors

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 2]
```diff
@@ -110,20 +119,17 @@ struct CorrSparse {
...
 
-  Filter trackFilter = (nabs(aod::track::eta) < cfgEtaCut) && (aod::track::pt > cfgPtCutMin) && (aod::track::pt < cfgPtCutMax) && ((requireGlobalTrackInFilter()) || (aod::track::isGlobalTrackSDD == (uint8_t) true))
-                       //&& (aod::track::tpcChi2NCl < cfgCutChi2prTPCcls)
-                       && (nabs(aod::track::dcaZ) > cfgDCAzCut) && (cfgDCAxySigmaCut * (0.0015f + 0.005f / npow(aod::track::pt, 1.1f)) < nabs(aod::track::dcaXY));
-  //
+  Filter trackFilter = (nabs(aod::track::eta) < cfgEtaCut) && (aod::track::pt > cfgPtCutMin) && (aod::track::pt < cfgPtCutMax) && ((requireGlobalTrackInFilter()) || (aod::track::isGlobalTrackSDD == (uint8_t) true));
 
   // define the filtered collisions and tracks
+
   using aodCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::CentFT0Cs>>;
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:130]: For this one, as suggested, you have to use `UpperCamelCase`, i.e. `aodCollisions`
And similar for the similar ones

**Review by victor-gonzalez — COMMENTED** [review]
Have a look at my suggestions

**Comment by Thorkj** [comment]
I have applied all the suggestions you mentioned, which removed all the linter issues except the name of the file itself. So I changed the name and the Source in the CMakeList.txt now this causes new issues... sorry for the inconvenience, is this is an expected consequence of changing the name?

**Comment by victor-gonzalez** [comment]
Be aware that your workflow name is correct
The part of `CMakeLists.txt` which corresponds to your task is correct
Other analyzers will take care of the other `linter` suggestions on that file

File: PWGCF/TwoParticleCorrelations/Tasks/corrSparse.cxx [hunk 3]
```diff
@@ -51,10 +51,15 @@ struct CorrSparse {
...
+    // Make histograms to check the distributions after cuts
     registry.add("deltaEta_deltaPhi_same", "", {HistType::kTH2D, {axisDeltaPhi, axisDeltaEta}}); // check to see the delta eta and delta phi distribution
     registry.add("deltaEta_deltaPhi_mixed", "", {HistType::kTH2D, {axisDeltaPhi, axisDeltaEta}});
+    registry.add("Phi", "Phi", {HistType::kTH1D, {{72, 0, TwoPI, "Phi"}}});
+    registry.add("Eta", "Eta", {HistType::kTH1D, {{40, -2, 2, "Eta"}}});
+    registry.add("pT", "pT", {HistType::kTH1D, {axisPtTrigger}});
+    registry.add("Nch", "N_{ch}", {HistType::kTH1D, {axisMultiplicity}});
 
     registry.add("Sparse_mixed", "", {HistType::kTHnSparseD, {{axisVertex, axisPtTrigger, axisPtAssoc, axisMultiplicity, axisDeltaPhi, axisDeltaEta}}}); // Make the output sparse
     registry.add("Sparse_same", "", {HistType::kTHnSparseD, {{axisVertex, axisPtTrigger, axisPtAssoc, axisMultiplicity, axisDeltaPhi, axisDeltaEta}}});
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/corrSparse.cxx:65]: I would recommend two thing in here
- do you really need `double`?
- have a look at the `CorrelationContainer` under `PWGCF/Core` and one of its usages within `PWGCF/Tasks/correlations.cxx` 

This histograms are going to be really big. The `double` to `float` directly half the size
The standard `THnSparse` uses two bins per dimension to store underflow and overflow which with six dimensions this is a significant amount of memory. The `CorrelationContainer` uses an `O2` version of `THnSparse` which does not incorporate the underflow and overflow bins

**Comment by victor-gonzalez** [comment]
> You added that the issues for the pull request is due to errors which others will fix, but I am currently under time constraints of a project needed to be handed in soon. Due to this I would like to ask if you know how long I can expect for this pull request to take. I also saw your comment regarding using CorrelationContainer, which I tried using in the beginning of my project(which caused a few problems), but was advised to just use sparses since the post processing uses normal sparses.

The issues in `CMakeLists.txt` will be fixed in the near future
They don't prevent your PR being merged and approved
Strange you had issues with the `CorrelationContainer`. I've used them to do a tutorial and a few checks a while ago, and there are analyses with approved results based on them
Regarding using sparses, if at some point you have memory issues, you know where to look at

You can proceed setting the PR as ready

**Review by victor-gonzalez — APPROVED** [review]

**Pull request merged**